File: Lab_2/task_1/1.go
package main

import (
	"fmt"
	"strconv"
)

func compressString(s string) string {
	var compressed string
	count := 1
	length := len(s)

	for i := 1; i <= length; i++ {
		if i < length && s[i] == s[i-1] {
			count++
		} else {
			compressed += string(s[i-1]) + strconv.Itoa(count)
			count = 1
		}
	}

	if len(compressed) < length {
		return compressed
	} else {
		fmt.Println("Сжатие не уменьшило строку. Возвращена исходная строка.")
		return s
	}
}

func main() {
	var s string
	fmt.Print("Введите строку: ")
	fmt.Scanln(&s)
	fmt.Println("Результат:", compressString(s))
}

File: Lab_2/task_1/1.php
<?php

function compressString($s) {
    $compressed = '';
    $count = 1;
    $length = strlen($s);

    for ($i = 1; $i <= $length; $i++) {
        if ($i < $length && $s[$i] === $s[$i - 1]) {
            $count++;
        } else {
            $compressed .= $s[$i - 1] . $count;
            $count = 1;
        }
    }

    if (strlen($compressed) < $length) {
        return $compressed;
    } else {
        echo "Сжатие не уменьшило строку. Возвращена исходная строка.\n";
        return $s;
    }
}

echo "Введите строку: ";
$s = trim(fgets(STDIN));
echo "Результат: " . compressString($s) . "\n";

?>
